core/direct-rpc-server/src/rpc_responder.rs

```diff
@@ -0,0 +1,290 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{DirectRpcError, DirectRpcResult, RpcConnectionRegistry, RpcHash, SendRpcResponse};
+use codec::{Decode, Encode};
+use itc_tls_websocket_server::WebSocketConnection;
+use itp_types::{DirectRequestStatus, RpcResponse, RpcReturnValue, TrustedOperationStatus};
+use log::*;
+use std::{sync::Arc, vec::Vec};
+
+pub struct RpcResponder<Registry, Hash, Connection>
+where
+	Registry: RpcConnectionRegistry<Hash = Hash, Connection = Connection>,
+	Hash: RpcHash,
+{
+	connection_registry: Arc<Registry>,
+}
+
+impl<Registry, Hash, Connection> RpcResponder<Registry, Hash, Connection>
+where
+	Registry: RpcConnectionRegistry<Hash = Hash, Connection = Connection>,
+	Hash: RpcHash,
+{
+	pub fn new(connection_registry: Arc<Registry>) -> Self {
+		RpcResponder { connection_registry }
+	}
+}
+
+impl<Registry, Hash, Connection> SendRpcResponse for RpcResponder<Registry, Hash, Connection>
+where
+	Registry: RpcConnectionRegistry<Hash = Hash, Connection = Connection>,
+	Hash: RpcHash,
+	Connection: WebSocketConnection,
+{
```

Comment:
These methods used to be in the untrusted worker part, as o-calls. They are now moved to the `direct-rpc-server` crate, and used from inside the enclave.